plugins/outputs/logzio/logzio.go

```diff
@@ -0,0 +1,176 @@
+
+  ## How often Logz.io sender should drain the queue.
+  ## Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+  # drain_duration = "3s"
+
+  ## Where Logz.io sender should store the queue
+  ## queue_dir = Sprintf("%s%s%s%s%d", os.TempDir(), string(os.PathSeparator),
+  ##                     "logzio-buffer", string(os.PathSeparator), time.Now().UnixNano())
+
+  ## Logz.io account token
+  token = "your logz.io token" # required
+
+  ## Use your listener URL for your Logz.io account region.
+  # url = "https://listener.logz.io:8071"
+`
+
+type Logzio struct {
+	CheckDiskSpace bool            `toml:"check_disk_space"`
+	DiskThreshold  int             `toml:"disk_threshold"`
+	DrainDuration  string          `toml:"drain_duration"`
+	Log            telegraf.Logger `toml:"-"`
+	QueueDir       string          `toml:"queue_dir"`
+	Token          string          `toml:"token"`
+	URL            string          `toml:"url"`
+
+	sender *lg.LogzioSender
+}
+
+type Metric struct {
+	Metric     map[string]interface{} `json:"metrics"`
+	Dimensions map[string]string      `json:"dimensions"`
+	Time       time.Time              `json:"@timestamp"`
+	Type       string                 `json:"type"`
+}
+
+func (l *Logzio) initializeSender() error {
+	if l.Token == "" || l.Token == "your logz.io token" {
+		return fmt.Errorf("token is required")
+	}
+
+	drainDuration, err := time.ParseDuration(l.DrainDuration)
+	if err != nil {
+		return fmt.Errorf("failed to parse drain_duration: %s", err)
+	}
+
+	diskThreshold := l.DiskThreshold
+	if diskThreshold < minDiskThreshold || diskThreshold > maxDiskThreshold {
+		return fmt.Errorf("threshold has to be between %d and %d", minDiskThreshold, maxDiskThreshold)
+	}
+
+	l.sender, err = lg.New(
+		l.Token,
+		lg.SetCheckDiskSpace(l.CheckDiskSpace),
+		lg.SetDrainDiskThreshold(l.DiskThreshold),
+		lg.SetDrainDuration(drainDuration),
+		lg.SetTempDirectory(l.QueueDir),
+		lg.SetUrl(l.URL),
+	)
+
+	if err != nil {
+		return fmt.Errorf("failed to create new logzio sender: %v", err)
+	}
+
+	l.Log.Infof("Successfuly created Logz.io sender: %s %s %s %d", l.URL, l.QueueDir,
+		l.DrainDuration, l.DiskThreshold)
+	return nil
+}
+
+// Connect to the Output
+func (l *Logzio) Connect() error {
+	l.Log.Debug("Connecting to logz.io output...")
+	return l.initializeSender()
+}
+
+// Close any connections to the Output
+func (l *Logzio) Close() error {
+	l.Log.Debug("Closing logz.io output")
+	l.sender.Stop()
+	return nil
+}
+
+// Description returns a one-sentence description on the Output
+func (l *Logzio) Description() string {
+	return logzioDescription
+}
+
+// SampleConfig returns the default configuration of the Output
+func (l *Logzio) SampleConfig() string {
+	return sampleConfig
+}
+
+// Write takes in group of points to be written to the Output
+func (l *Logzio) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	l.Log.Debugf("Recived %d metrics", len(metrics))
```

Comment:
I didn't include a test for the Write() because most of it is tested in the logziosender.go file. I'll remove logziosender and add a simple HTTP client + tests :)

Reply:
@ssoroka done